controller.js

```diff
@@ -39,32 +43,49 @@ const getComments = (req, res) => {
       res.status(400).send({ msg: "Invalid article_id" });
     });
 };
-// post comments
+// Post comments
 const commentPost = (req, res) => {
-    const { article_id } = req.params;
-    const reqBody = req.body;
-    commentPostModel(article_id, reqBody).then((result) => {
-      res.status(201).send({ comment: result });
-    })
+  const { article_id } = req.params;
+  const reqBody = req.body;
+  if (!reqBody.username || !reqBody.body) {
+    return res.status(400).send({ msg: "Invalid username provided" });
+  }
 
-  
+  commentPostModel(article_id, reqBody).then((result) => {
+    res.status(201).send({ comment: result });
+  });
 };
-
-// get articles by id
+// Get articles by id
 const getArticleId = (req, res) => {
   const { article_id } = req.params;
+      if (isNaN(article_id)) {
+        return res.status(400).send({ msg: "Invalid article_id provided" });
+      }
   getArticalsbyId(article_id)
     .then((result) => {
       if (!result) {
         return res.status(404).send({ msg: "articles not found" });
       }
       res.status(200).send({ articles: result });
     })
-    .catch(() => {
-      if (isNaN(article_id)) {
-        res.status(404).send({ msg: "Invalid article_id provided" });
+};
+//PATCH /api/articles/:article_id
+const updateArticleVotes = (req, res, next) => {
+  const { article_id } = req.params;
+  const { inc_votes } = req.body;
+  if (typeof inc_votes !== "number") {
+    return res
+      .status(400)
+      .send({ msg: "Bad Request: inc_votes must be a number" });
+  }
```

Comment:
You dont have to manually check the above - let psql deal with it. It will error if the data type is incorrect